dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ClarinEPersonImportController.java

```diff
@@ -0,0 +1,151 @@
+            throws AuthorizeException, SQLException {
+        Context context = obtainContext(request);
+        if (Objects.isNull(context)) {
+            throw new RuntimeException("Context is null!");
+        }
+
+        //get eperson attribute values
+        String selfRegisteredString = request.getParameter("selfRegistered");
+        boolean selfRegistered = getBooleanFromString(selfRegisteredString);
+        String lastActiveString = request.getParameter("lastActive");
+        Date lastActive;
+        try {
+            lastActive = getDateFromString(lastActiveString);
+        } catch (ParseException e) {
+            throw new RuntimeException("Cannot import eperson, because the last_active is entered incorrectly!");
+        }
+        //salt and digest_algorithm are changing with password
+        EPersonRest epersonRest = ePersonRestRepository.createAndReturn(context);
+        EPerson eperson = ePersonService.find(context, UUID.fromString(epersonRest.getUuid()));
+        eperson.setSelfRegistered(selfRegistered);
+        eperson.setLastActive(lastActive);
+        ePersonService.update(context, eperson);
+
+        String hasUserRegistrationString = request.getParameter("userRegistration");
+        boolean userRegistration = getBooleanFromString(hasUserRegistrationString);
+
+        //create user registration if exists
+        if (userRegistration) {
+            String organization = request.getParameter("organization");
+            String confirmationString = request.getParameter("confirmation");
+            boolean confirmation = getBooleanFromString(confirmationString);
+
+            ClarinUserRegistration clarinUserRegistration = new ClarinUserRegistration();
+            clarinUserRegistration.setOrganization(organization);
+            clarinUserRegistration.setConfirmation(confirmation);
+            clarinUserRegistration.setEmail(eperson.getEmail());
+            clarinUserRegistration.setPersonID(eperson.getID());
+            clarinUserRegistrationService.create(context, clarinUserRegistration);
+        }
+        epersonRest = converter.toRest(eperson, utils.obtainProjection());
+        context.complete();
+
+        return epersonRest;
+    }
+
+    /**
+     * Convert String value to boolean.
+     * @param value input value
+     * @return converted input value to boolean
+     */
+    private boolean getBooleanFromString(String value) {
+        boolean output = false;
+        if (StringUtils.isNotBlank(value)) {
+            output = Boolean.parseBoolean(value);
+        }
+        return output;
+    }
+
+    /**
+     * Convert String value to Date.
+     * Expects two possible date formats, but more can be added.
+     * @param value
+     * @return converted input value to Date
+     * @throws ParseException if parse error
+     */
+    private Date getDateFromString(String value) throws ParseException {
+        Date output = null;
+        if (StringUtils.isNotBlank(value)) {
```

Comment:
isBlank do not continue

Reply:
exactly, stop processing if `value` is blank